docs/dev/Lua API.rst

```diff
@@ -5766,82 +5766,134 @@ The names of the functions are also available as the keys of the
+  :gear_tiles:        Optional array of 2 or 4 indexes. First two define ascii tiles and next two graphics tiles.
+                      This overrides default gear tiles.
 
-    :canBeRoomSubset:
-        a flag if this building can be counted in room. 1 means it can, 0 means it can't and -1 default building behaviour
-    :auto_gears:
-        a flag that automatically fills up gears and animations. It looks over the building definition for gear icons and maps them.
+* ``setOnUpdate(workshop_type,interval,callback)``
 
-    Animate table also might contain:
+  Setup callback to be called every ``interval`` of ticks for each building of this type. Note: low interval
+  numbers and/or many workshops that use this might reduce DF performance.
 
-    :frameLength:
-        how many ticks does one frame take OR
-    :isMechanical:
-        a bool that says to try to match to mechanical system (i.e. how gears are turning)
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :interval:        how many ticks to skip between event triggers
+  :callback:        function to call. Function signature is ``func(workshop)`` where ``workshop`` is of type
+                    ``df.building_workshopst``
 
-``getPower(building)`` returns two number - produced and consumed power if building can be modified and returns nothing otherwise
+* ``getPower(building)``
 
-``setPower(building,produced,consumed)`` sets current power production and consumption for a building.
+  Returns two number - produced and consumed power if building can be modified and returns nothing otherwise.
+
+  :building:   specific workshop that produces or consumes power
+
+* ``setPower(building, power_consumed, power_produced)``
+
+  Sets current power production and consumption for a specific workshop building. Can be used to make buildings that
+  dynamically change power consumption and production.
+
+  :building:   specific workshop that produces or consumes power
+  :power_consumed:   set building to consume this amount of power
+  :power_produced:   output this amount of power
+
+
+Events
+------
+
+This module exports two events. However only one is documented here and is intended to be used directly. To use
+``onUpdateAction`` instead call ``setOnUpdate`` function.
+
+* ``onSetTriggerState(workshop,state)``
+
+  Notify when building is triggered from linked lever or trap.
+
+  :workshop:   object of type ``df.building_workshopst`` that is triggered.
+  :state:   integer value of new state.
 
 Examples
 --------
 
 Simple mechanical workshop::
 
-  require('plugins.building-hacks').registerBuilding{name="BONE_GRINDER",
-    consume=15,
-    gears={x=0,y=0}, --connection point
-    animate={
-      isMechanical=true, --animate the same conn. point as vanilla gear
-      frames={
-      {{x=0,y=0,42,7,0,0}}, --first frame, 1 changed tile
-      {{x=0,y=0,15,7,0,0}} -- second frame, same
+  local bld=require('plugins.building-hacks')
```

Comment:
```suggestion
  local bhacks = require('plugins.building-hacks')
```

Reply:
`bld` is consistently used throughout DFHack to refer to "current building". using the name to refer to the module would be confusing